app/Repositories/AuthorRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Models\Author;
+use App\Http\Resources\AuthorResource;
+use Illuminate\Support\Facades\Response;
+use App\Repositories\Interfaces\AuthorInterface;
+
+class AuthorRepository implements AuthorInterface
+{
+
+    //Show All Authors
+    public function allAuthors()
+    {
+        return AuthorResource::collection(Author::all());
+    }
+
+    //Create New Author
+    public function createAuthors($data)
+    {
+        return new AuthorResource(Author::create($data->all()));
+    }
+
+    //Update an Author
+    public function updateAuthor($data, $id)
+    {
+        $author = new AuthorResource(Author::findOrFail($id));
```

Comment:
we will not put author resource here

Reply:
fixed, using author resouce on return statement now